samplers/src/main/java/io/opentelemetry/contrib/sampler/RuleBasedRoutingSampler.java

```diff
@@ -60,7 +62,14 @@ public SamplingResult shouldSample(
       return fallback.shouldSample(parentContext, traceId, name, spanKind, attributes, parentLinks);
     }
     for (SamplingRule samplingRule : rules) {
-      String attributeValue = attributes.get(samplingRule.attributeKey);
+      String attributeValue;
+      if (samplingRule.attributeKey.getKey().equals(THREAD_ID.getKey())) {
+        attributeValue = String.valueOf(Thread.currentThread().getId());
```

Comment:
let's remove support for `thread.id` since it's not a string attribute (and probably no one wants to sample on it anyways)